src/main/java/org/json/JSONObject.java

```diff
@@ -276,6 +276,21 @@ public JSONObject(JSONTokener x) throws JSONException {
      *            If a key in the map is <code>null</code>
      */
     public JSONObject(Map<?, ?> m) {
+      this(m, 0, new JSONParserConfiguration());
+    }
+
+    public JSONObject(Map<?, ?> m, JSONParserConfiguration jsonParserConfiguration) {
+        this(m, 0, jsonParserConfiguration);
+    }
+
+    /**
+     * Construct a JSONObject from a map with recursion depth.
+     *
+     */
+    protected JSONObject(Map<?, ?> m, int recursionDepth, JSONParserConfiguration jsonParserConfiguration) {
```

Comment:
Should be private, I think.

Reply:
@stleary 
For above case I have kept the accessor as private.
And as for other cases where we are passing `recursionDepth` as a parameter, I've updated the accessor to default visibility, limiting access to the current packages. This adjustment ensures that there is no need for broader accessibility.